### PR TITLE
feat: error resilient array with different character length

### PR DESCRIPTION
**Pull request by parth121101**
Can't really figure out what has just happened. Why test references of other files updated. 
@Pranavchiku can you help me here.

### CODE REVIEW

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 1]
```diff
@@ -125,6 +125,7 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                 } else {
                     tmp = nullptr;
                     tmp_vec.clear();
+                    return;
```
Comment (Pranavchiku) [inline src/lfortran/semantics/ast_body_visitor.cpp:128]: because of these I suspect

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 2]
```diff
@@ -133,6 +134,7 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                     diag.add(e.d);
                     tmp = nullptr;
                     tmp_vec.clear();
+                    return;
```
Comment (Pranavchiku) [inline src/lfortran/semantics/ast_body_visitor.cpp:137]: and this

Comment (parth121101) [reply src/lfortran/semantics/ast_body_visitor.cpp]: Thanks for the help here. It worked

File: tests/reference/run-array_constructor_with_different_char_lengths_cc-746cb03.stdout [hunk 3]
```diff
@@ -0,0 +1,3 @@
+ball
+ball1
```
Comment (Pranavchiku) [inline tests/reference/run-array_constructor_with_different_char_lengths_cc-746cb03.stdout:2]: this is interesting. Why did it print these?

**Review by Pranavchiku — APPROVED** [review]
I think this is good, let's wait for one more approval.

File: tests/reference/run-array_constructor_with_different_char_lengths_cc-746cb03.stdout (hunk 3 above)
Comment (HarshitaKalani) [reply tests/reference/run-array_constructor_with_different_char_lengths_cc-746cb03.stdout:2]: +1

**Review by HarshitaKalani — APPROVED** [review]
I'm also curious about this:https://github.com/lfortran/lfortran/pull/5328#discussion_r1835649736

Otherwise this looks good to me too.

**Comment by certik** [comment]
This PR doesn't add any new fix or implementation. It only adds tests. Do we need more tests right now?

It does expose a bug though (the spurious prints), so that has to be fixed separately.

Right now let's only add tests to test new functionality, not current functionality unless we have to, since it slows down our testing runs. So we should spare our "testing budget" for new features. Once we reach beta, then let's add thorough tests for every compiler error (but not as one test per error, but probably like 100 errors per file, all reported at once, for testing performance reasons).

**Comment by parth121101** [comment]
> This PR doesn't add any new fix or implementation. It only adds tests. Do we need more tests right now?
> 
> It does expose a bug though (the spurious prints), so that has to be fixed separately.
> 
> Right now let's only add tests to test new functionality, not current functionality unless we have to, since it slows down our testing runs. So we should spare our "testing budget" for new features. Once we reach beta, then let's add thorough tests for every compiler error (but not as one test per error, but probably like 100 errors per file, all reported at once, for testing performance reasons).

Yes @certik . We will add a single test file to test more errors. And regarding the bug present here, I would work on fixing it in this PR itself.

**Comment by certik** [comment]
Yes. In fact once we have one file with errors, then I think you can add more error messages tests (like in this PR), as the overhead should be minimal.

**Review by Pranavchiku — APPROVED** [review]

**Pull request merged**